file-proxy/auth-file-proxy/auth-file-proxy.js

```diff
@@ -0,0 +1,136 @@
+            headers: { 'content-type': 'application/json' },
+            body: JSON.stringify({
+              query: `
+              mutation {
+                createMyFile (secret: "${result.secret}",
+                              name: "${result.name}",
+                              size: ${result.size},
+                              contentType: "${result.contentType}" ,
+                              url: "${result.url.replace('files.graph.cool', req.headers.host + '/' + webtaskName)}",
+                              fileId: "${result.id}")
+                {
+                  name
+                  size
+                  url
+                  id
+                  contentType
+                }
+              }`
+            })
+          },
+          (err,resp,body) => {
+            const response = JSON.parse(body);
+            const newId = response.data.createMyFile == null ? null : response.data.createMyFile.id;
+
+            if (newId == null && response.errors[0].code == '3008')
+            {
+              // If the user is not allowed to create a MyFile node, we return '403 Forbidden'
+              // The uploaded file will be cleaned up by our watcher
+              res.status(403).send();
+            }
+            else {
+              // Return the response body to the client, just like the 'normal' File API.
+              res.status(200).send(response.data.createMyFile);
+            }
+          }).auth(null, null, true, req.token == undefined ? null : req.token);
+      });
+    });
+
+  // Let multiparty parse the incoming request
+  form.parse(req);
+});
+
+// The download endpoint
+app.get('/:projectid/:filesecret', (req, res) => {
+
+  // The decryption method, using the same secret password
+  const decipher = crypto.createDecipher('aes256', req.webtaskContext.secrets.FILE_ENC_PASSWORD);
+
+  // First, we read the MyFile node with the user authentication to get the original file URL
+  request.post(
+    {
+      url: `https://api.graph.cool/simple/v1/${req.params.projectid}`,
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify({
+        query: `query { MyFile(secret: "${req.params.filesecret}") { id file { url } } }`
+      })
+    },
+    (err,resp,body) => {
+      const response = JSON.parse(body);
+      const newId = response.data.MyFile == null ? null : response.data.MyFile.id;
+
+      if (newId == null && response.errors[0].code == '3008')
+      {
+        // If the user is not allowed to create a MyFile node, we return '403 Forbidden'
+        // The uploaded file will be cleaned up by our watcher
+        res.status(403).send();
+        return;
+      }
+
+      // The request to the actual file
+      var resource = request.get(response.data.MyFile.file.url);
+
+      // As soon as we get a response, we copy the headers
+      // Content-length is removed, because the decrypted file does not have the same length
```

Comment:
is it feasible to adjust the `content-length` header with the new length, instead?